asdf/external_reference.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import, division, unicode_literals, print_function
+
+from .asdftypes import AsdfType
+
+
+class ExternalArrayReference(AsdfType):
+    """
+    Store a reference to an array in an external File.
+
+    Parameters
+    ----------
+
+    filepath: `str`
+        The path to the path to be referenced. Can be relative to the file
+        containing the reference.
+
+    target: `object`
+        Some internal target to the data in the file. Examples may include a HDU
+        index, a HDF path or an asdf fragment.
+
+    dtype: `str`
+        The (numpy) dtype of the contained array.
+
+    shape: `tuple`
+        The shape of the array to be loaded.
+    """
+    name = "core/externalarray"
+
```

Comment:
We need an explicit `version` field here as well. For now it should just be `1.0.0`.